internal/cri/server/sandbox_update_resources.go

```diff
@@ -0,0 +1,51 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package server
+
+import (
+	"context"
+	"fmt"
+
+	runtime "k8s.io/cri-api/pkg/apis/runtime/v1"
+
+	"github.com/containerd/containerd/v2/pkg/tracing"
+	"github.com/containerd/log"
+)
+
+func (c *criService) UpdatePodSandboxResources(ctx context.Context, r *runtime.UpdatePodSandboxResourcesRequest) (*runtime.UpdatePodSandboxResourcesResponse, error) {
+	span := tracing.SpanFromContext(ctx)
+	sandbox, err := c.sandboxStore.Get(r.GetPodSandboxId())
+	if err != nil {
+		return nil, fmt.Errorf("failed to find sandbox: %w", err)
+	}
+
+	span.SetAttributes(tracing.Attribute("sandbox.id", sandbox.ID))
+
+	overhead := r.GetOverhead()
+	resources := r.GetResources()
+	err = c.nri.UpdatePodSandboxResources(ctx, &sandbox, overhead, resources)
+	if err != nil {
+		return nil, fmt.Errorf("NRI sandbox update failed: %w", err)
+	}
+
```

Comment:
Should we not store these fields in the sandbox store?  if we don't store these values in the sandbox store... we won't have a way to sync them with the plugin for restart scenarios of the runtime/plugin or for a late binding nri plugin.

Reply:
@klihub I suppose if we do sync the resources update we could do it after each pod sync..

Reply:
Updated to store the `Overhead` and `Resources` in the `StatusStorage` and update in this handler. For container updates it looks like this the store is [updated in the NRI API client](https://github.com/containerd/containerd/blob/v2.0.2/internal/cri/nri/nri_api_linux.go#L427-L436), should we be doing something similar here instead of in this handler?

Reply:
interesting... I prefer the way you have it here over saving in the nri api client and I suppose sans an NRI config change the results should be the same. NRI could be off for a set of long running pods... then on after an edit of the config and a restart. Might not affect expectations though.

Reply:
> @klihub I suppose if we do sync the resources update we could do it after each pod sync..

@mikebrow Sorry Mike, I'm not sure how to interpret that. 

> Should we not store these fields in the sandbox store? if we don't store these values in the sandbox store... we won't have a way to sync them with the plugin for restart scenarios of the runtime/plugin or for a late binding nri plugin.

@mikebrow @chrishenzie So shouldn't an `UpdatePodSandboxResources` in the end update both the in-memory and stored pod state so that the overall state of the pod becomes indistinguishable from if it was started with the updated resources in the first place ? Then if an NRI plugin is started after the update, it will only see the pod in its updated state.